orttraining/orttraining/core/session/training_session.cc

```diff
@@ -330,6 +332,10 @@ Status TrainingSession::ConfigureForTraining(
   ORT_RETURN_IF_ERROR(BuildGradientGraph(
       weight_names_to_train_copy, loss_name, config.gradient_graph_config, *session_logger_));
 
+  if (config.pipeline_config.has_value() && config.pipeline_config.value().do_partition) {
+    ORT_RETURN_IF_ERROR(RemoveNonDifferentiableEdgeInPartition(model_->MainGraph(), non_differentiable_edge_names));
```

Comment:
We may still be making this call even without non-diff edges (and so trip the assertion added above)?  I'd tend toward keeping the API simple, and handling the empty case inside Remove*